core/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -346,6 +355,19 @@ protected void processWay(ReaderWay way) {
             }
         }
 
+        List<CustomArea> customAreas = estimatedCenter == null ? emptyList() : customAreaIndex.query(estimatedCenter.lat, estimatedCenter.lon);
+        CountryRule countryRule = null;
+        for (CustomArea customArea : customAreas) {
+            Object countryCode = customArea.getProperties().get("ISO3166-1:alpha3");
+            if (countryCode != null) {
+                NewCountry country = NewCountry.valueOf(countryCode.toString());
```

Comment:
Doesn't `Enum.valueOf()` yield an `IllegalArgumentException` if it can't find the value?

```suggestion
                NewCountry country = NewCountry.find(countryCode.toString());
```

Reply:
Yes, it does. For country this should be ok, because I will move the countries.geojson into the source bundle and the NewCountry enum should then always be in sync with the country codes used in this file. But otherwise I agree it would be more useful to return `null` or something like a 'missing' value in case some value is missing (like `road_class=my_404_class`).